File: modules/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/request/VoidBody.java
package com.roxiemobile.networkingapi.network.rest.request;

import com.roxiemobile.networkingapi.network.http.MediaType;
import com.roxiemobile.networkingapi.network.rest.HttpBody;

public class VoidBody implements HttpBody
{
// MARK: - Construction

    private VoidBody() {
        // Do nothing
    }

// MARK: - Methods

    @Override
    public MediaType mediaType() {
        return null;
    }

    @Override
    public byte[] body() {
        return null;
    }

}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/http/body/VoidBody.kt
@file:Suppress("unused")

package com.roxiemobile.networkingapi.network.http.body

import com.roxiemobile.networkingapi.network.http.MediaType

class VoidBody: HttpBody {

// MARK: - Construction

    private constructor() {
        throw UnsupportedOperationException()
    }

// MARK: - Methods

    override val mediaType: MediaType
        get() = throw UnsupportedOperationException()

    override val body: ByteArray
        get() = throw UnsupportedOperationException()
}
